llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -5130,39 +5162,51 @@ bool X86InstrInfo::optimizeCompareInstr(MachineInstr &CmpInstr, Register SrcReg,
     default:
       llvm_unreachable("Unreachable!");
     case X86::SUB64rm:
+    case X86::SUB64rm_ND:
       NewOpcode = X86::CMP64rm;
```

Comment:
Will it be a problem if `SUB64rm_ND` is assigned with EGPR reigsters?

Reply:
Do we need to consider `SUB64rm_EVEX`?

Reply:
No, it's a redundant encoding and never generated by compiler.